eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/ReadRepoInfoFromSourceManifest.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.UnifiedBuild.Tasks
+{
+    // Takes a path to a source-manifest.json file and
+    // reads the information for a specific repo from it.
+    public class ReadRepoInfoFromSourceManifest : Task
+    {
+        [Required]
+        public string SourceManifest { get; set; } = "";
```

Comment:
@premun Thoughts on using the source manifest or the V.D.xml file for this? We should ideally be going in one direction to eventually converge.

Reply:
Going to merge this as more people reached out to me asking for the feature but I'm glad to address more feedback in a follow-up change of course.